src/js/service/movie/model.js

```diff
@@ -1,16 +1,34 @@
-"use strict"
+const MOVIE = 'MOVIE'
+const TV_SHOW = 'TV_SHOW'
 
-class Movie {
-    constructor(type, name, pictureUrl, score, matchingScore) {
+class Show {
+    // you could set group the optional parameters and make it explicit
+    // that they are optional.
+    // something like
+    constructor(type, name, {pictureUrl = '', score = null, matchingScore = null}) {
```

Comment:
required params as a params, optional as an additional object. Could put everything in an object as well. As you prefer.